pkg/services/screenshot/screenshot.go

```diff
@@ -235,13 +171,86 @@ func (s *ObservableScreenshotService) Take(ctx context.Context, opts ScreenshotO
 
 	screenshot, err := s.service.Take(ctx, opts)
 	if err != nil {
-		defer s.failures.Inc()
+		if errors.Is(err, models.ErrDashboardNotFound) {
+			defer s.failures.With(prometheus.Labels{
+				"reason": "dashboard_not_found",
+			}).Inc()
+		} else if errors.Is(err, context.Canceled) {
+			defer s.failures.With(prometheus.Labels{
+				"reason": "context_canceled",
+			}).Inc()
+		} else {
+			defer s.failures.With(prometheus.Labels{
+				"reason": "error",
+			}).Inc()
+		}
 	} else {
 		defer s.successes.Inc()
 	}
 	return screenshot, err
 }
 
+// RemoteRenderScreenshotService takes screenshots using a remote render service.
+type RemoteRenderScreenshotService struct {
+	ds dashboards.DashboardService
+	rs rendering.Service
+}
+
+func NewRemoteRenderScreenshotService(ds dashboards.DashboardService, rs rendering.Service) ScreenshotService {
+	return &RemoteRenderScreenshotService{
+		ds: ds,
+		rs: rs,
+	}
+}
+
+// Take returns a screenshot or an error if either the dashboard does not exist or
+// the service failed to screenshot the dashboard. It uses both the context and the
+// timeout in ScreenshotOptions, however the context is used in any database queries
+// and the request to the remote render service, while the timeout in ScreenshotOptions
+// is passed to the remote render service where it is used as a client timeout. It is
+// not recommended to pass a context without a deadline.
+func (s *RemoteRenderScreenshotService) Take(ctx context.Context, opts ScreenshotOptions) (*Screenshot, error) {
+	q := models.GetDashboardQuery{Uid: opts.DashboardUID}
+	if err := s.ds.GetDashboard(ctx, &q); err != nil {
+		return nil, err
+	}
+
+	u := url.URL{}
+	u.Path = path.Join("d-solo", q.Result.Uid, q.Result.Slug)
+	p := u.Query()
+	p.Add("orgId", strconv.FormatInt(q.Result.OrgId, 10))
+	p.Add("panelId", strconv.FormatInt(opts.PanelID, 10))
+	u.RawQuery = p.Encode()
+
+	opts = opts.SetDefaults()
+	renderOpts := rendering.Opts{
+		AuthOpts: rendering.AuthOpts{
+			OrgID:   q.Result.OrgId,
+			OrgRole: models.ROLE_ADMIN,
+		},
+		ErrorOpts: rendering.ErrorOpts{
+			ErrorConcurrentLimitReached: true,
+			ErrorRenderUnavailable:      true,
+		},
+		TimeoutOpts: rendering.TimeoutOpts{
+			Timeout: opts.Timeout,
+		},
+		Width:           opts.Width,
+		Height:          opts.Height,
+		Theme:           opts.Theme,
+		ConcurrentLimit: setting.AlertingRenderLimit,
+		Path:            u.String(),
+	}
+
+	result, err := s.rs.Render(ctx, renderOpts, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to take screenshot: %w", err)
+	}
+
+	screenshot := Screenshot{Path: result.FilePath}
+	return &screenshot, nil
+}
+
```

Comment:
Nit: It is hard to see the difference beteween added code and removed code. I would recommend doing that either in two commits or do not move the code at all.